hub/subscribe.go

```diff
@@ -87,16 +87,21 @@ func (h *Hub) initSubscription(w http.ResponseWriter, r *http.Request) (*Subscri
 	authorizedAlltargets, authorizedTargets := authorizedTargets(claims, false)
 	subscriber := NewSubscriber(authorizedAlltargets, authorizedTargets, topics, rawTopics, templateTopics, retrieveLastEventID(r))
 
-	if subscriber.LastEventID != "" {
-		h.sendMissedEvents(w, r, subscriber)
-	}
-
 	// Create a new channel, over which the hub can send can send updates to this subscriber.
 	updateChan := make(chan *serializedUpdate)
 
 	// Add this client to the map of those that should
 	// receive updates
+
+	useHistory := subscriber.LastEventID != ""
+	if useHistory {
+		h.history.Lock()
+		h.sendMissedEvents(w, r, subscriber)
+	}
 	h.newSubscribers <- updateChan
+	if useHistory {
+		h.history.Unlock()
```

Comment:
are you sure that at this point the new subscriber is fully registered in the hub and pending writes will be dispatched to it?